src/views/PurchaseOrderReview.vue

```diff
@@ -51,13 +51,21 @@
           </ion-item>
         </div>
       </div>
-      <div v-if="segmentSelected === 'all'">
-        <PurchaseOrderDetail :purchaseOrders="purchaseOrders" :itemsByPoId ="purchaseOrders.parsed"  />
+      <div v-if="searchedProduct?.pseudoId && queryString">
+        <PurchaseOrderDetail :purchaseOrders="purchaseOrders" :itemsByPoId ="{[searchedProduct?.orderId]: [searchedProduct]}"  />
       </div>
-      <div v-for="(po, poId) in purchaseOrders.parsed" :key="poId" >
-        <PurchaseOrderDetail v-if="segmentSelected === poId" :itemsByPoId="{[poId]: po}" :purchaseOrders="purchaseOrders" />
+
+      <div v-else>
+        <div v-if="segmentSelected === 'all'">
+          <PurchaseOrderDetail :purchaseOrders="purchaseOrders" :itemsByPoId ="purchaseOrders.parsed"  />
+        </div>
+        <div v-for="(po, poId) in purchaseOrders.parsed" :key="poId" >
+          <PurchaseOrderDetail v-if="segmentSelected === poId" :itemsByPoId="{[poId]: po}" :purchaseOrders="purchaseOrders" />
```

Comment:
Searching pseudoId and changing segments is not honouring the PO segment to search and is searching on all POs. For instance, if I have 2 POs and I search for something from PO1 in all segments and change the segment to PO2, the searched product will appear there from PO1.

Reply:
Fixed.

Reply:
Fixed.